plugins/bil/bil_main.ml

```diff
@@ -0,0 +1,104 @@
+  Bil.register_pass name ~desc (Stmt.normalize ~normalize_exp)
+
+let bnf1 =
+  register_norm "bnf1" "Transforms BIL to the first normalized form" false
+let bnf2 =
+  register_norm "bnf2" "Transforms BIL to the second normalized form" true
+
+let normalizations =
+  [ "0", [];
+    "1", [bnf1];
+    "2", [bnf2]; ]
+
+let fold_consts =
+  Bil.register_pass "constant-folding" ~desc:"constant-folding" Bil.fold_consts
+
+let consts_propagation =
+  Bil.register_pass "constant-propagation" ~desc:"constant-propagation" Bil.propagate_consts
+
+let prune_dead =
+  Bil.register_pass "prune-dead"
+    ~desc:"dead code elimination for virtual variables" Bil.prune_dead
+
+let o1 = [fold_consts; consts_propagation; prune_dead]
+
+let optimizations =
+  [ "0", [];
+    "1", o1; ]
+
+let print_passes () =
+  Bil.passes () |> List.map ~f:Bil.Pass.to_string |> List.sort ~cmp:String.compare |>
+  List.iter ~f:(printf "%s\n");
+  exit 0
+
+let pass = Config.enum
+    (List.map (Bil.passes ()) ~f:(fun p -> Bil.Pass.to_string p, p))
+
+let () =
+  let () = Config.manpage [
+      `S "DESCRIPTION";
+      `P "Controls the BIL transformation pipeline.";
+
+      `P "The BIL transformation pipeline is used to transform the BIL
+        code of each machine instruction and is applied to the output
+        of a lifter. Each piece of transformation is called $(i,a pass).
+        This plugin enables full control over the pass selection as
+        well as provides some predefined passes. See the
+        $(b,--bil-list-passes) for a comprehensive list of available
+        transformations. The pipeline is formed as a concatenation of
+        all normalization passes that correspond to the selected level
+        of normalization (setting it to zero will disable normalizations),
+        all optimization passes applicable to the selected level of
+        optimization (again the zero level corresponds to an empty
+        list of optimizations), and all passes selected via the
+        $(b,--bil-passes) command line argument. All passes are
+        applied until a fixed point is reached (i.e., until the
+        outcome of the pipeline is equal to the income) or until a
+        loop is detected. Thus to prevent infinite looping each
+        registered pass should be a pure function of its inputs (same
+        input shall result in the same output).";
+
+      `P "A new pass could be registered via the $(b,Bil.register_pass)
+        function. It won't be automatically selected, so it should be
+        specified explicitly via the $(b,--bil-passes) command line
+        argument."
+    ] in
+  let norml =
+    let doc = "Specifies the BIL normalization level.
+      The normalization process doesn't change the semantics of
+      a BIL program, but applies some transformations to simplify it.
+      There are two BIL normal forms (bnf): bnf1 and bnf2, both
+      of which are described in details in $(b,bap.mli).
```

Comment:
It should be a reference to documentation not an mli file, i.e., `$(b,bap)(3)`, also add the SEE ALSO section with the reference to BAP, and bapdoc will make it clickable.